### PR TITLE
[JENKINS-58956] - Enable PCT and whitelist java.util.Collections$EmptyMap

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Includes #46 from @fcojfernandez.
Parent POM update is not complete (there is a mess reported by enforcer, needs to be fixed eventually).

According to code inspection JENKINS-58956 happens only when you migrate data for old configs/builds. I didn't find a quick way to test it, so here is just a quickfix till https://github.com/jenkinsci/jenkins/pull/3234 gets released.

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees @imod @jglick

File: pom.xml [hunk 1]
```diff
@@ -34,10 +34,11 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.javadoc.skip>true</maven.javadoc.skip>
-        <jenkins.version>1.642.3</jenkins.version>
-        <jenkins-test-harness.version>2.18</jenkins-test-harness.version>
+        <jenkins.version>2.7.3</jenkins.version>
```
Comment (jglick) [inline pom.xml]: Do you want to make `Jenkinsfile` test against 2.102 also?

Comment (oleg-nenashev) [reply pom.xml]: Nope. It blows up on enforcer

**Comment by oleg-nenashev** [comment]
@reviewbybees done
@imod would it be possible to ship it while we decide what to do with the core release?

**Comment by imod** [comment]
@oleg-nenashev sure, I'll get it out soon

**Comment by oleg-nenashev** [comment]
thanks!

**Comment by imod** [comment]
@oleg-nenashev done: https://wiki.jenkins.io/display/JENKINS/Config+File+Provider+Plugin#ConfigFileProviderPlugin-Version2.17(17.Jan.2018)

**Comment by oleg-nenashev** [comment]
Thanks!

**Pull request merged**